apps/dot_ext/loggers.py

```diff
@@ -0,0 +1,199 @@
+from django.db.utils import IntegrityError
+from oauth2_provider.models import get_application_model
+from .models import AuthFlowUuid
+
+"""
+  Logger related functions for dot_ext/mymedicare_cb modules and auth flow trace logging.
+"""
+
+SESSION_AUTH_FLOW_TRACE_KEYS = ['auth_uuid', 'auth_client_id', 'auth_app_id', 'auth_app_name', 'auth_pkce_method']
+
+
+def cleanup_session_auth_flow_trace(request=None):
+    '''
+    Function for cleaning up auth flow related items
+    in session.
+    '''
+    if request:
+        # We are done using auth flow trace values, clear them from the session.
+        for k in SESSION_AUTH_FLOW_TRACE_KEYS:
+            try:
+                del request.session[k]
+            except KeyError:
+                pass
+
+
+def create_session_auth_flow_trace(request):
+    '''
+    Function for creating auth flow log tracing related items.
+
+    - Creates a new AuthFlowUuid instance.
+    - Sets new auth flow values in session.
+
+    Returns an AuthFlowUuid instance or None.
+    '''
+    if request:
+        application = None
+
+        # Create new authorization flow trace UUID.
+        new_auth_uuid = str(uuid.uuid4())
+
+        request.session['auth_uuid'] = new_auth_uuid
+
+        client_id_param = request.GET.get("client_id", None)
+        auth_pkce_method = request.GET.get("code_challenge_method", None)
+
+        if client_id_param:
+            # Get the application.
+            Application = get_application_model()
+            try:
+                application = get_application_model().objects.get(client_id=client_id_param)
+
+                # Set values in session.
+                auth_flow_dict = {"auth_uuid": new_auth_uuid,
+                                  "auth_app_id": str(application.id),
+                                  "auth_app_name": str(application.name),
+                                  "auth_client_id": str(application.client_id),
+                                  "auth_pkce_method": str(auth_pkce_method),
+                                  }
+                set_session_auth_flow_trace(request, auth_flow_dict)
+            except Application.DoesNotExist:
+                # Clear values in session. Set to empty value to denote not found.
+                auth_flow_dict = {"auth_uuid": new_auth_uuid,
+                                  "auth_app_id": "",
+                                  "auth_app_name": "",
+                                  "auth_client_id": "",
+                                  "auth_pkce_method": "",
+                                  }
+                set_session_auth_flow_trace(request, auth_flow_dict)
+                application = None
+
+        if application:
+            # Create and return AuthFlowUuid instance for tracking.
+            AuthFlowUuid.objects.create(auth_uuid=new_auth_uuid,
+                                        client_id=application.client_id,
+                                        auth_pkce_method=auth_pkce_method)
```

Comment:
You could do the same thing here and move this into the try statement because application will always be None otherwise.